src/sentry/utils/performance_issues/performance_detection.py

```diff
@@ -673,7 +675,7 @@ def is_creation_allowed_for_organization(self, organization: Organization) -> bo
         )
 
     def is_creation_allowed_for_project(self, project: Project) -> bool:
-        return True  # Detection always allowed by project for now
+        return self.settings["detection_rate"] > random.random()
```

Comment:
A bit of a nit, but the name of this method is misleading now since it's more of a sampling decision at this point rather than whether creation is allowed

Reply:
Mm I don't know about that! `self.settings["detection_rate"]` is from project settings, after all! This nicely mirrors the `for_system` and `for_organization` methods, which also make sampling decisions. The fact that the project parameter is not used definitely a little weird, but IMO incidental! Other detectors might choose to use it, but this one happens not to

Reply:
I see your point though! Like, if it _fetched project settings in this method_ based on the project, that sure would make a lot more sense

Reply:
Agreed, that would be much clearer! But I don't mind it as much since `detection_rate` is technically from project settings as you explained 😄